### PR TITLE
Add post detail

### PR DESCRIPTION
**Pull request by jasonmule**
This pull request makes the following changes:
-

Test these changes by:
- Navigating to post details for a post and check that:
   - Post detail actions work as expected
   - Add to survey adds a new post to the post's current survey
   - Post content and values are laid out correctly

Fixes ushahidi/platform#1055, fixes ushahidi/platform#1088.

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/221)

### CODE REVIEW

File: app/post/controllers/post-detail-controller.js [hunk 1]
```diff
@@ -274,4 +277,33 @@ function (
             Notify.showApiErrors(errorResponse);
         });
     };
+
+    $scope.addToSurvey = function () {
+       $location.path('/posts/create/' + post.form.id);
+    };
+
+    // Get duration since post was last updated
+    function getUpdateDuration() {
```
Comment (rjmackay) [inline app/post/controllers/post-detail-controller.js]: Moment can do this for you: moment().fromNow(); .. it also does the locatization for you more sanely.
post card directive wraps that with a switcher to change to long date format for things that happened a while ago
```
function formatDates() {
                var created = moment($scope.post.created),
                    now = moment();

                if (now.isSame(created, 'day')) {
                    $scope.displayTime = created.fromNow();
                } else {
                    $scope.displayTime = created.format('LL');
                }
                $scope.displayTimeFull = created.format('LLL');
            }
```
Something similar to that would be better than hacking our own together.

File: app/post/directives/post-detail-actions-directive.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            $scope.showDropDown = false;
+            $scope.toggleDropDown = function () {
+                $scope.showDropDown = !$scope.showDropDown;
+            };
+
+            $scope.addToCollection = function (post) {
+                $rootScope.$emit('collectionToggle:show', [post]);
+            };
+
+            $scope.delete = function (post) {
```
Comment (rjmackay) [inline app/post/directives/post-detail-actions-directive.js]: Can we share this code with the delete function in the listing?
Probably move it to something a PostActionsService. The confirm + notify part of the pattern is repeated, only the redirect is different.

File: app/post/directives/post-detail-actions-directive.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return {
+        restrict: 'E',
+        replace: true,
+        scope: {
+            post: '='
+        },
+        templateUrl: 'templates/posts/post-detail-actions.html',
+        link: function ($scope) {
+            $scope.showDropDown = false;
+            $scope.toggleDropDown = function () {
```
Comment (rjmackay) [inline app/post/directives/post-detail-actions-directive.js]: why doesn't this just use the existing dropdown directives?

File: server/www/templates/posts/detail.html [hunk 4]
```diff
@@ -1,125 +1,147 @@
...
-                        <h4 ng-if="mapDataLoaded" translate>post.location</h4>
-                        <div ng-if="mapDataLoaded">
-                            <leaflet id="post-map" class="map" defaults="defaults" geojson="geojson" center="center" layers="layers"></leaflet>
-                          </div>
-                    </div>
-                </div>
+<div class="mode-context" ng-controller="navigation as nav">
+
+  <header class="mode-context-header has-logo">
+    <h1 class="mode-context-title"><a href="/" ng-bind="nav.site.name"></a></h1>
```
Comment (rjmackay) [inline server/www/templates/posts/detail.html]: I think this is supposed to be the form title, not site title (that should go in a breadcrumb)

File: server/www/templates/posts/detail.html [hunk 5]
```diff
@@ -1,125 +1,147 @@
...
+
+  <span class="mode-context-trigger">
+    <svg class="iconic">
+      <use xlink:href="../img/iconic-sprite.svg#chevron-bottom"></use>
+    </svg>
+    <span class="label hidden">Show more/less</span>
+  </span>
+
+  <div class="mode-context-body">
+    <p ng-bind="nav.site.description"></p>
```
Comment (rjmackay) [inline server/www/templates/posts/detail.html]: I don't think site description is what we're supposed to be showing here.. check with @brandonrosage but I'd guess it should be the form description..

File: app/post/controllers/post-detail-controller.js (hunk 1 above)
Comment (jasonmule) [inline app/post/controllers/post-detail-controller.js]: Thanks!

**Comment by jasonmule** [comment]
@rjmackay I did not get around to using the dropdown service but the other changes should be in, including messages and form UI for unstructured posts.

**Comment by rjmackay** [comment]
Reviewed 1 of 12 files at r1, 15 of 18 files at r3.
Review status: 16 of 23 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[app/post/directives/post-detail-actions-directive.js, line 31 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/221#-KJXtlvsHX2pC_rCDRxK-r2-31:-KJm8gv5TE0kBkVV9-AH:262468467) ([raw file](https://github.com/ushahidi/platform-client/blob/cb596ee67dc0f0f3928b26ba5db31a873df0e3c4/app/post/directives/post-detail-actions-directive.js#L31)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Can we share this code with the delete function in the listing?
> 
> Probably move it to something a PostActionsService. The confirm + notify part of the pattern is repeated, only the redirect is different.
</details>
:thumbsup: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/221)*

**Pull request merged**